app/core/ai_cache.py

```
@@ -0,0 +1,106 @@
+from collections import OrderedDict
+
+@dataclass
+class CacheItem:
+    value: Any
+    created_at: datetime
+    ttl: Optional[int] = None
+    weight: float = 1.0
+    access_count: int = 0
+
+class AIOptimizedCache:
+    def __init__(self, max_size: int = 1000, redis_url: Optional[str] = None):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)